File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/UpdateRoundsUseCase.java
package es.etologic.mahjongscoring2.domain.use_cases;

import java.util.List;

import javax.inject.Inject;

import es.etologic.mahjongscoring2.data.repositories.GamesRepository;
import es.etologic.mahjongscoring2.data.repositories.RoundsRepository;
import es.etologic.mahjongscoring2.domain.model.GameWithRounds;
import es.etologic.mahjongscoring2.domain.model.Round;
import io.reactivex.Single;

public class UpdateRoundsUseCase {

    private final RoundsRepository roundsRepository;
    private final GamesRepository gamesRepository;

    @Inject
    public UpdateRoundsUseCase(RoundsRepository roundsRepository, GamesRepository gamesRepository) {
        this.roundsRepository = roundsRepository;
        this.gamesRepository = gamesRepository;
    }

    public Single<List<Round>> getGameRounds(long gameId) {
        return roundsRepository.getRoundsByGame(gameId);
    }

    public Single<GameWithRounds> addRound(Round round) {
        return roundsRepository.insertOne(round)
                .flatMap(aLong -> gamesRepository.getOneWithRounds(round.getGameId()));
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/UpdateRoundsUseCase.kt
package es.etologic.mahjongscoring2.domain.use_cases

import es.etologic.mahjongscoring2.data.repositories.GamesRepository
import es.etologic.mahjongscoring2.data.repositories.RoundsRepository
import es.etologic.mahjongscoring2.domain.model.GameWithRounds
import es.etologic.mahjongscoring2.domain.model.Round
import io.reactivex.Single
import javax.inject.Inject

class UpdateRoundsUseCase @Inject
constructor(private val roundsRepository: RoundsRepository, private val gamesRepository: GamesRepository) {
    
    fun getGameRounds(gameId: Long): Single<List<Round>> {
        return roundsRepository.getRoundsByGame(gameId)
    }
    
    fun addRound(round: Round): Single<GameWithRounds> = roundsRepository.insertOne(round)
        .flatMap { gamesRepository.getOneWithRounds(round.gameId.toLong()) } //FixMe: ¿ToLong???
}
